### ISSUE TITLE
[BUG] `RuntimeError: Resource temporarily unavailable` when using option `-e` to check expired date

### ISSUE THREAD

**Issue by hiddify1** [dashingsoft/pyarmor#1176]
I receive the following exception in some systems:

```
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 187, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/usr/lib/python3.10/runpy.py", line 146, in _get_module_details
    return _get_module_details(pkg_main_name, error)
  File "/usr/lib/python3.10/runpy.py", line 110, in _get_module_details
    __import__(pkg_name)
  File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/__init__.py", line 2, in <module>
    from .pyarmor_runtime_000000 import __pyarmor__
  File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/pyarmor_runtime_000000/__init__.py", line 2, in <module>
    from .pyarmor_runtime import __pyarmor__
RuntimeError: Resource temporarily unavailable (1:10934)
```
```
pyarmor gen -r -i -O . -e 60   package

INFO     Python 3.10.11
INFO     Pyarmor 8.1.6 (trial), 000000, non-profits
INFO     Platform linux.x86_64
INFO     search inputs ...

```

is it because 3.10.11 is not compatible with 3.10.6??

**Comment by jondy** [comment]
It can't connect to nts server to check expire date. It can't be fixed by pyarmor, what pyarmor could do is to refine error message to something like `can not connect to nts server`.

**Comment by jondy** [comment]
The reason it's that device can't connect to ntp server to check expire date. It's network problem, it may work next time if device could connect to internet. It can't be fixed by pyarmor, what pyarmor could do is to refine error message to something like `can not connect to nts server`. 

Any idea if can't not connect to nts server?

**Comment by hiddify1** [comment]
Great thanks however everything is working except this
Maybe the ntp is not accessible 

Maybe it is better to check multiple ntp instead of one

**Comment by hiddify1** [comment]
which ntp service are you using?

**Comment by hiddify1** [comment]
maybe using such a services also be good to avoid problem when some isps block NTP port
https://worldtimeapi.org/api/timezone/Europe/Paris

**Comment by jondy** [comment]
By default pyarmor use `pool.ntp.org`, it could be configured by
```
pyarmor cfg nts "any.ntp.org"
```

But it only handle NTP protocol.

I would not change the default check method.

If want to use other service like https://worldtimeapi.org/api/timezone/Europe/Paris, create runtime hook to verify it by users.

For example, a hook script may be like this 
```python
def check_expire_date():
    from http.client import HTTPSConnection
    expire = __pyarmor__(1, None, b'keyinfo', 1)
    host = 'worldtimeapi.org'
    path = '/api/timezone/Europe/Paris'
    conn = HTTPSConnection(host)
    conn.request("GET", path)
    res = conn.getresponse()
    if res.code == 200:
        data = res.read()
        s = data.find(b'"unixtime":')
        n = data.find(b',', s)
        current = int(data[s+11:n])
        if current > expire:
            raise RuntimeError('license is expired')
        return True
if not check_expire_date():
    raise RuntimeError('something is wrong')
``` 

Then generate script with local expire date
```
pyarmor gen -e .30 foo.py
```

The runtime hook will be implemented in 8.2, it will be embedded into obfuscated scripts and executed in module level

**Comment by jondy** [comment]
Implemented in 8.2.0, document it here
https://pyarmor.readthedocs.io/en/v8.2/tutorial/customization.html#using-hook-to-check-network-time-by-other-service
